src/MelodyGenerator.js

```diff
@@ -3,18 +3,122 @@ const MelodyQueue = require('./MelodyQueue.js');
 const waveGenerator = require('./WaveGenerator.js');
 const note = require('./note.js');
 
-class MelodyGenerator
-{
+class MelodyGenerator {
     fromString(melodyString) {
         // put your code here
+        this.defines = [];
+        this.sounds = [];
+        let step1 = melodyString.replace(new RegExp(', ', "g"), ',').split("\n");
+
+        for (let i = 0; i < step1.length; i++) {
+            this.parse(step1[i]);
+        }
+
         let melody = new MelodyQueue();
-        melody.enqueueTone(durations.Q, note('A4'));
+        this.buildMelody(melody, this.sounds);
+
         return melody;
     }
 
     getNotesArray(notes) {
         return notes.map((n) => note(n));
     }
+
+
+    //pare a line
+    parse(line) {
+        let command = line.split(' ');
+
+        switch (command[0]) {
+            case 'play':
+                this.parseCommandPlay(command);
+                break;
+            case 'define':
+                this.parseCommandDefine(command);
+                break;
+
+            default:
+                break;
```

Comment:
in future - maybe show warning is a better idea

Reply:
Why not...